index_bidding_history1.php

```diff
@@ -0,0 +1,134 @@
+<?php include 'determine_result.php'?>
+
+
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.2/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-Zenh87qX5JnK2Jl0vWa8Ck2rdkQ2Bzep5IDxbcnCeuOxjzrPF/et3URy9Bv1WTRi" crossorigin="anonymous">
+    <title>Bidding history lists</title>
+    
+</head>
+
+<body>
+<nav class="navbar navbar-expand-lg bg-light">
+  <div class="container-fluid">
+    <a class="navbar-brand" href="#" style="color:blue;">Hello!  
+<?php $login_email = $_SESSION['email'] ;
+  $userNameSQL = "SELECT * from users where users.email= '$login_email' "; 
+	      $result = mysqli_query($db_conn, $userNameSQL);
+	      $row = mysqli_fetch_assoc($result);
+        echo $row["firstname"] . " " . $row["lastname"];
+  //  if ( false===$result ) {
+  //       printf("error: %s\n", mysqli_error($db_conn));
+  //       }
+  //       else {
+  //       echo 'done.';
+  //       }
+  ?>  
+        </a>
+    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
+      <span class="navbar-toggler-icon"></span>
+    </button>
+    <div class="collapse navbar-collapse" id="navbarNav">
+      <ul class="navbar-nav">
+        <li class="nav-item">
+          <a class="nav-link active" aria-current="page" href="/onlineauction/index.php">Home</a>
+        </li>
+        <li class="nav-item">
+          <a class="nav-link" href="#">Previous Page</a>
+        </li>
+        <li class="nav-item">
+          <a class="nav-link" href="/onlineauction/logout.php">logout</a>
+        </li>
+      </ul>
+    </div>
+  </div>
+</nav>
+
+
+    <div class="container-fuild">
+    
+        <table class="table">
+            <thead class="table-dark">
+              <tr><h1 style="text-align:center">Bidding history lists</h1></tr>
+                <tr>
+                    <th style="width:5%"> Bid ID</th>
+                    <th style="width:10%"> Product Name</th>
+                    <th> Product ID</th>
+                    <th style="width:15%"> Picture</th>
+                    <th> Bidding Status</th>
+                    <th> Your price</th>
+                    <th> Current Highest price</th>
+                    <th> Bidding Result</th>
+                    <th> Bidding End Date</th>
+                </tr>
+            </thead>
+            <tbody>
+              <?php
+                 $mail = $_SESSION['email'] ;
+                 $sql = "SELECT * FROM bidding JOIN item ON bidding.item_ID = item.item_ID JOIN users ON users.user_ID = bidding.buyer_ID WHERE users.user_type IN ('buyer', 'both') AND users.email = '$mail' order by item.item_id, bidding.bid_id";  # select data from sql table
```

Comment:
@yly828 There's a redundancy in this query. I have tried cutting off something, and still get the same result.
`[SELECT](http://localhost:8888/phpMyAdmin5/url.php?url=https://dev.mysql.com/doc/refman/5.7/en/select.html) * FROM bidding JOIN item ON bidding.item_ID = item.item_ID JOIN users ON users.user_ID = bidding.buyer_ID WHERE users.email = 'ucabat4@ucl.ac.uk' order by item.item_id;`